08/classes_homework.py

```diff
@@ -0,0 +1,74 @@
+# Napiš si třídy pro cokoliv chceš tak, aby splňovaly zadání:
+# Dvě (nebo více) odvozených tříd.
+# Jedna odvozená třída bude kompletně přepisovat metodu nadřazené třídy.
+# Druhá odvozená třída bude rozšiřovat metodu nadřazené třídy pomocí super().
+# Obě odvozené třídy budou mít stejnou metodu, která bude dělat stejnou věc
+# jiným způsobem (koťátko mňouká, štěňátko štěká).
+
+
+class Printers:
+    def __init__(self, name):
+        self.name = name
+
+    def printing_material(self):
+        '''Vyhodnocuje mnozstvi tiskarskeho materialu (naplne)'''
+        return self.printing_material > 0
+
+    def production(self, product):
+        '''Vypise, jaky produkt tiskarna tiskne.'''
+        print(f'{self.name} is printing {product}...')
```

Comment:
Pokud jsem správně odhadl, co za [doménovou logiku](https://stackoverflow.com/questions/360860/what-is-domain-logic) tyhle třídy mají reprezentovat:

1. Že existují různé druhy tiskáren
2. Všechny druhy tiskáren potřebují _nějakou náplň_ a stav náplně nějak drží v atributu `self.printing_material` _(přestože společný základ tiskáren - třída `Printers` - tenhle atribut má jako metodu, ale to je asi bugísek)_
3. Některé druhy tiskáren k tisku také potřebují papír - a stav dostupnosti papíru se drží v atributu `self.empty_paper`.
4. Tiskárny, pokud mají nějakou náplň a papír (pokud daná tiskárna papír vůbec potřebuje), umí tisknout to, co do nich pošlu přes metodu `self.production(...)`

... tak bych asi v téhle metodě `production(...)` očekával, že před `print(f'{self.name} is printing {product}...')` bude nějaký IF, který nejdřív zkontroluje, zda tiskárna má vůbec nějakou náplň a _(pokud tiskárna používá papír)_ taky nějaký papír 🖨️  _(pokud jo, bude se tisknout - zde asi reprezentováno voláním funkce `print` - pokud ne, tak to `print()` volat nebude, nebo by to klidně mohlo vyhodit výjimku - např. `raise RuntimeError("Unable to print")`_

Ale tady se nic nekontroluje a prostě se tiskne - i když zrovna třeba není na co - a nebo ani čím 😛